pkg/storageos/webhook.go

```diff
@@ -60,47 +65,60 @@ func (s Deployment) createWebhookService() error {
 // webhooks.  The configuration will be updated by the api-manager to include
 // the CA cert bundle.
 func (s Deployment) createMutatingWebhookConfiguration() error {
-	failurePolicy := admissionv1.Ignore
-	matchPolicy := admissionv1.Exact
-	sideEffects := admissionv1.SideEffectClassNoneOnDryRun
-	svcPort := webhookPort
 	scopeAll := admissionv1.AllScopes
-	path := "/mutate-pods"
-
 	webhooks := []admissionv1.MutatingWebhook{
-		{
-			Name: webhookPodMutatorName,
-			ClientConfig: admissionv1.WebhookClientConfig{
-				Service: &admissionv1.ServiceReference{
-					Name:      WebhookServiceName,
-					Namespace: s.stos.Spec.GetResourceNS(),
-					Port:      &svcPort,
-					Path:      &path,
+		s.mutatingWebhookConfiguration(webhookPodMutatorName, webhookPort, mutatePodPath, []admissionv1.RuleWithOperations{
+			{
+				Operations: []admissionv1.OperationType{admissionv1.Create},
+				Rule: admissionv1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"pods"},
+					Scope:       &scopeAll,
 				},
 			},
-			Rules: []admissionv1.RuleWithOperations{
-				{
-					Operations: []admissionv1.OperationType{admissionv1.Create},
-					Rule: admissionv1.Rule{
-						APIGroups:   []string{""},
-						APIVersions: []string{"v1"},
-						Resources:   []string{"pods"},
-						Scope:       &scopeAll,
-					},
+		}),
+		s.mutatingWebhookConfiguration(webhookPVCMutatorName, webhookPort, mutatePVCPath, []admissionv1.RuleWithOperations{
+			{
+				Operations: []admissionv1.OperationType{admissionv1.Create},
+				Rule: admissionv1.Rule{
+					APIGroups:   []string{""},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"persistentvolumeclaims"},
+					Scope:       &scopeAll,
 				},
 			},
-			FailurePolicy:           &failurePolicy,
-			MatchPolicy:             &matchPolicy,
-			NamespaceSelector:       &metav1.LabelSelector{},
-			SideEffects:             &sideEffects,
-			AdmissionReviewVersions: []string{"v1"},
-		},
+		}),
 	}
 	labels := podLabelsForAPIManager(s.stos.Name)
 
 	return s.k8sResourceManager.MutatingWebhookConfiguration(MutatingWebhookConfigName, labels, webhooks).Create()
 }
 
+func (s Deployment) mutatingWebhookConfiguration(name string, port int32, path string, rules []admissionv1.RuleWithOperations) admissionv1.MutatingWebhook {
```

Comment:
I wonder if we need to set ports. It works fine with the default 443 and 9443. I don't see only the port changing in the future. I think there's a high possibility of the webhook service to change as well.
So, maybe add both or none for now and add them in the future when we need to change them.

Reply:
I've removed the port for now - it's not something we support changing, so we can revisit later if needed.